megamek/src/megamek/client/ui/swing/tileset/TilesetManager.java

```diff
@@ -0,0 +1,556 @@
+    private HashMap<ArrayList<Integer>, EntityImage> mechImages = new HashMap<ArrayList<Integer>, EntityImage>();
+
+    // hex images
+    private HexTileset hexTileset;
+
+    private Image minefieldSign;
+    private Image nightFog;
+
+    /** An opaque hex shape used to limit draw operations to the exact hex shape. */
+    private Image hexMask;
+
+    private Image artilleryAutohit;
+    private Image artilleryAdjusted;
+    private Image artilleryIncoming;
+    
+    /**
+     * Hexes under the effects of ECM have a shaded "static" image displayed,
+     * to represent the noise generated by ECM.  This is a cache that stores
+     * images for various colors (for Players, and possibly multiple players
+     * in the same hex).
+     */
+    private HashMap<Color, Image> ecmStaticImages = new HashMap<Color, Image>();
+    
+    /** Creates new TilesetManager. */
+    public TilesetManager(BoardView1 bv) throws IOException {
+        boardview = bv;
+        hexTileset = new HexTileset(boardview.game);
+        tracker = new MediaTracker(boardview);
+        try {
+            camos = new DirectoryItems(
+                    Configuration.camoDir(),
+                    "", //$NON-NLS-1$
+                    ImageFileFactory.getInstance()
+            );
+        } catch (Exception e) {
+            camos = null;
+        }
+        mechTileset.loadFromFile("mechset.txt"); //$NON-NLS-1$
+        wreckTileset.loadFromFile("wreckset.txt"); //$NON-NLS-1$
+        try {
+            hexTileset.incDepth = 0;
+            hexTileset.loadFromFile(PreferenceManager.getClientPreferences().getMapTileset());
+        } catch (Exception FileNotFoundException) {
+            System.out.println("Error loading tileset, "
+                    + "reverting to default hexset! " + "Could not find file: "
+                    + PreferenceManager.getClientPreferences().getMapTileset());
+            if (new MegaMekFile(Configuration.hexesDir(), FILENAME_DEFAULT_HEX_SET).getFile().exists()){
+                hexTileset.loadFromFile(FILENAME_DEFAULT_HEX_SET);
+            } else {
+                System.err.println("Critical Error loading default tileset!");
+            }
+        }
+        PreferenceManager.getClientPreferences().addPreferenceChangeListener(this);
+        GUIPreferences.getInstance().addPreferenceChangeListener(this);
+    }
+
+    /** React to changes in the settings. */
+    public void preferenceChange(PreferenceChangeEvent e) {
+        // A new Hex Tileset has been selected
+        if (e.getName().equals(IClientPreferences.MAP_TILESET)) {
+            HexTileset hts = new HexTileset(boardview.game);
+            try {
+                hexTileset.incDepth = 0;
+                hts.loadFromFile((String) e.getNewValue());
+                hexTileset = hts;
+                boardview.clearHexImageCache();
+            } catch (IOException ex) {
+                return;
```

Comment:
Is there a chance that something meaningful could happen here? If so, let's log it.